apps/ewallet/lib/ewallet/trackers/transaction_tracker.ex

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWallet.TransactionTracker do
+  @moduledoc """
+  This module is a GenServer started dynamically for a specific eWallet transaction
+  It will registers itself with the blockchain adapter to receive events about
+  a given transactions and act on it
+  """
+  use GenServer, restart: :temporary
+  require Logger
+
+  alias EWallet.BlockchainTransactionState
+  alias ActivityLogger.System
+
+  # TODO: handle failed transactions
+
+  def start_link(state) do
+    GenServer.start_link(__MODULE__, state)
+  end
+
+  def init(%{transaction: transaction} = state) do
+    adapter = Application.get_env(:ewallet, :blockchain_adapter)
+    :ok = adapter.subscribe(:transaction, transaction.blockchain_tx_hash, self())
+    {:ok, state}
+  end
+
+  def handle_cast(
+        {:confirmations_count, tx_hash, confirmations_count},
+        %{transaction: transaction} = state
+      ) do
+    case transaction.blockchain_tx_hash == tx_hash do
+      true ->
+        adapter = Application.get_env(:ewallet, :blockchain_adapter)
+        threshold = Application.get_env(:ewallet, :blockchain_confirmations_threshold)
+
+        if is_nil(threshold) do
+          Logger.warn("Blockchain Confirmations Threshold not set in configuration: using 10.")
+        end
+
+        update_confirmations_count(
+          adapter,
+          state,
+          confirmations_count,
+          confirmations_count > (threshold || 10)
```

Comment:
Can we define a module attribute for the default confirmations count (10) somewhere?

Reply:
I also wonder isn't it suppose to be `>=`?